packages/commonwealth/client/scripts/hooks/api/useChains.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import { Topic } from 'models';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import $ from 'jquery';
+import app from 'state';
+import queryKeys from './queryKeys';
+
+const useChains = () => {
+  // auth state to come from zustand, localstorage is just for demo purposes
+  const JWT = localStorage.getItem('jwt');
+
+  const queryClient = useQueryClient();
+  const [chainIdToFetch, setChainIdToFetch] = useState<string>();
+
+  const fetchBulkOffChainQuery = useQuery({
+    queryKey: [queryKeys.BULK_OFFCHAIN(chainIdToFetch)],
+    queryFn: async () => {
+      // api call
+      const [response] = await Promise.all([
+        $.get(`${app.serverUrl()}/bulkOffchain`, {
+          community: null,
+          chain: chainIdToFetch,
+          jwt: JWT,
+        }),
+      ]);
+
+      // process payload
+      const { topics } = response.result;
+
+      // set topics state (changes are propogated to all components that use the useTopics hook)
+      await queryClient.cancelQueries({
+        queryKey: [queryKeys.TOPICS_FOR_CHAIN(chainIdToFetch)],
+      });
+      queryClient.setQueryData(
+        [queryKeys.TOPICS_FOR_CHAIN(chainIdToFetch)],
+        (t: []) => [...(t || []), ...topics].map((x) => new Topic(x))
+      );
+
```

Comment:
This is how we would update cache state of a resource in one hook from another hook

Reply:
What's with the cancelQueries / setQueryData stuff?

Reply:
RQ will auto set the data from api in cache, setQueryData is used to manually do that when the data we want to store has a different schema. In the above case, RQ already has cache data as array and we are storing a new object in that array.

RQ can auto retry/refetch queries after the cache time expires, `cancelQueries` is a fail safe in situation when we are modifying the cache manually and want to prevent the scheduled api call from overriding our manual update attempt.